packages/react-components/src/components/time-sync/timeSelection.tsx

```diff
@@ -52,31 +56,97 @@ const messages: ViewportMessages = {
  * parent TimeSync and affect all other viewports in
  * that TimeSync group.
  */
-export const TimeSelection = () => {
+export const TimeSelection = ({ isPaginationEnabled }: { isPaginationEnabled?: boolean }) => {
   const { viewport, setViewport } = useViewport();
 
+  const getDisableForwardBoolean = (viewport: Viewport | undefined) => {
+    if (!viewport) return true;
+    if ('duration' in viewport) return true;
+    const duration = new Date(viewport.end).getTime() - new Date(viewport.start).getTime();
+    return new Date(viewport.end).getTime() + duration > Date.now();
+  };
+
   const handleChangeDateRange: NonCancelableEventHandler<DateRangePickerProps.ChangeDetail> = (event) => {
     const { value } = event.detail;
+    if (!value) return;
+    setViewport(dateRangeToViewport(value), 'date-picker');
+  };
 
+  const handlePaginateForward = () => {
+    const value = viewportToDateRange(viewport);
     if (!value) return;
+    if (value.type == 'absolute') {
+      const duration = new Date(value.endDate).getTime() - new Date(value.startDate).getTime();
+      const newEnd = new Date(new Date(value.endDate).getTime() + duration);
+      setViewport(
+        dateRangeToViewport({ startDate: value.endDate, endDate: newEnd.toISOString(), type: value.type }),
+        'date-picker'
+      );
+    }
+  };
 
-    setViewport(dateRangeToViewport(value), 'date-picker');
+  const handlePaginateBackward = () => {
+    const value = viewportToDateRange(viewport);
+    if (!value) return;
+    if (value.type == 'absolute') {
+      const duration = new Date(value.endDate).getTime() - new Date(value.startDate).getTime();
+      const newStart = new Date(new Date(value.startDate).getTime() - duration);
```

Comment:
edge case: does it handle it gracefully when the `startDate` has a value of the oldest available date?

Reply:
Thought about that. UTC starts at Jan 1st, 1970 and the UTC date object can go all the way back to 1900. Might be out of reach by customer pagination.